GVFS/GVFS.Virtualization/Projection/GitIndexProjection.FolderData.cs

```diff
@@ -49,6 +52,36 @@ public void Include()
                 }
             }
 
+            public string HashedChildrenNamesSha()
+            {
+                using (HashAlgorithm hash = SHA1.Create())
+                {
+                    byte[] sha = new byte[20];
+
+                    for (int i = 0; i < this.ChildEntries.Count; i++)
+                    {
+                        if (!this.ChildEntries[i].IsFolder)

```

Comment:
Since we are only using the `Name` now that is in the base class `FolderEntryData` and we can remove this check and use the property of the base class without casting and having 2 code paths.